internal/resource/resource_plugin_test.go

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) F5, Inc.
+//
+// This source code is licensed under the Apache License, Version 2.0 license found in the
+// LICENSE file in the root directory of this source tree.
+
+package resource
+
+import (
+	"context"
+	"testing"
+
+	"github.com/nginx/agent/v3/api/grpc/mpi/v1"
+	"github.com/nginx/agent/v3/internal/bus"
+	"github.com/nginx/agent/v3/internal/resource/resourcefakes"
+	"github.com/nginx/agent/v3/test/protos"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestResource_Process(t *testing.T) {
+	ctx := context.Background()
+
+	updatedInstance := &v1.Instance{
```

Comment:
is this useful to be stored in github.com/nginx/agent/v3/test/protos? If it's only used here maybe not

Reply:
I think its only used here for now. Would moving it if we end up using it else where be ok and leave it as is for now? I don't mind either way.